cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/mtom/MtomRequestWrapper.java

```diff
@@ -0,0 +1,106 @@
+package nl.nn.adapterframework.extensions.cmis.mtom;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import javax.mail.BodyPart;
+import javax.mail.internet.MimeMultipart;
+import javax.servlet.ServletInputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.entity.ContentType;
+import org.apache.logging.log4j.Logger;
+import org.springframework.mock.web.DelegatingServletInputStream;
+
+import nl.nn.adapterframework.http.InputStreamDataSource;
+import nl.nn.adapterframework.http.mime.MultipartEntityBuilder;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class MtomRequestWrapper extends HttpServletRequestWrapper {
+	protected Logger log = LogUtil.getLogger(this);
+
+	private HttpEntity entity;
+	private static final ContentType MTOM_XOP = ContentType.parse("application/xop+xml; charset=UTF-8");
+
+	public MtomRequestWrapper(ServletRequest request) throws IOException {
+		this((HttpServletRequest) request);
+	}
+
+	public MtomRequestWrapper(HttpServletRequest request) throws IOException {
+		super(request);
+
+		String contentType = super.getHeader("content-type");
+		if("POST".equalsIgnoreCase(request.getMethod())) {
+			try {
+				if(log.isTraceEnabled()) log.trace("found message with ContentType ["+contentType+"]");
+				boolean isMultipartRequest = contentType.contains("multipart");
+				MultipartEntityBuilder multipart = MultipartEntityBuilder.create();
+				multipart.setMtomMultipart();
+
+				if(isMultipartRequest) { // Multiple parts we need to iterate over
+					InputStreamDataSource dataSource = new InputStreamDataSource(contentType, super.getInputStream());
+					MimeMultipart mp = new MimeMultipart(dataSource);
```

Comment:
Hier lees je denk ik het hele request in je geheugen in. Zou dat te vermijden zijn?

Reply:
Misschien wel, misschien niet.